src/openassetio-core/include/openassetio/BatchElementError.hpp

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+#pragma once
+
+#include <string>
+#include <utility>
+
+#include <openassetio/errors.h>
+#include <openassetio/export.h>
+
+namespace openassetio {
+inline namespace OPENASSETIO_CORE_ABI_VERSION {
+/**
+ * Structure representing per-element batch operation errors.
+ *
+ * Many OpenAssetIO API functions take multiple inputs, i.e. a batch of
+ * elements, in order to allow the @ref manager backend to optimise bulk
+ * queries. The results of such queries are usually returned
+ * element-by-element via a callback.
+ *
+ * It is possible for the whole batch to fail due to some catastrophic
+ * error, in which case a standard exception workflow is expected.
```

Comment:
[consider] Based on our chat earlier about BAL, I was looking into HTTP status codes. They separate [Client Errors - 4xx](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses) from [Server Errors - 5xx](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses). This feels like the axis we've been grasping for when trying to work out what is a `BatchElementError` and what is an exception. Might be good to add this as a concrete example here as to when to use `BatchElementError`?

Reply:
HTTP status codes as an analogy is a bit tenuous, but a good rule of thumb.  Added a note in c1373efd, how is that?